### PR TITLE
middlewares

### PR DESCRIPTION
**Pull request by srwilliamg**
## Summary

- Added library express-validator
- Added schema for post request in **users/**
- Changed validations in users controller

## Trello Card

https://trello.com/c/A6VFpaow/27-middlewares

### CODE REVIEW

File: app/controllers/users.js [hunk 1]
```diff
@@ -1,24 +1,23 @@
...
-  if (!body) throw errors.badRequestError('Not data sent');
-  if (!name) throw errors.badRequestError('Not valid name');
-  if (!lastName) throw errors.badRequestError('Not valid last name');
-  if (!password || password.length < 8) throw errors.badRequestError('Not valid password');
-  if (!email || !email.includes(WOLOX_EMAIL_DOMAIN)) throw errors.badRequestError('Not valid email');
+  if (isValidRequest.length) {
+    throw errors.badRequestError(isValidRequest[0].msg);
+  }
 
   const createdUser = await UsersServices.createUser(body);
```
Comment (blueJayWays) [inline app/controllers/users.js:16]: you should use async/await inside try...catch blocks, especially when you call services that can fail.

File: app/middlewares/validators/users.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  last_name: {
+    in: ['body'],
+    notEmpty: false
+  },
+  email: {
+    in: ['body'],
+    isEmail: {
+      bail: true,
+      errorMessage: 'Should be an email'
```
Comment (blueJayWays) [inline app/middlewares/validators/users.js]: These error messages must be in the  constants file, then import them where are necessary, like in this validation, and the tests.

File: test/data/users.js [hunk 3]
```diff
@@ -5,8 +5,8 @@ module.exports = {
     email: 'william.salazar@wolox.co',
     password: 'thisIsAVerySecurePassword'
   },
-  NotValidNameMessage: 'Not valid name',
-  NotValidPasswordMessage: 'Not valid password',
+  NotValidNameMessage: 'Name cannot be null or empty',
+  NotValidPasswordMessage: 'Password length should be minimum 8',
```
Comment (blueJayWays) [inline test/data/users.js]: These error messages should be imported from the same constant of the validation middleware.

**Review by blueJayWays — CHANGES_REQUESTED** [review]

File: app/controllers/users.js (hunk 1 above)
Comment (srwilliamg) [reply app/controllers/users.js:16]: The controller is used in a function called requestHandler this function call the controller inside a try catch block

file: routes.js
app.post('/users', checkSchema(userPostSchema), requestHandler(UsersController.createUser));

File: app/middlewares/validators/users.js (hunk 2 above)
Comment (srwilliamg) [reply app/middlewares/validators/users.js]: done

File: test/data/users.js (hunk 3 above)
Comment (srwilliamg) [reply test/data/users.js]: done

**Pull request merged**